pandas/io/xml.py

```diff
@@ -507,6 +623,70 @@ def _parse_nodes(self) -> list[dict[str, str | None]]:
 
         return dicts
 
+    def _iterparse_nodes(self) -> list[dict[str, str | None]]:
+        from lxml.etree import iterparse
+
+        dicts: list[dict[str, str | None]] = []
+        row: dict[str, str | None] | None = None
+
+        if not isinstance(self.iterparse, dict):
+            raise TypeError(
+                f"{type(self.iterparse).__name__} is not a valid type for iterparse"
+            )
+
+        row_node = next(iter(self.iterparse.keys())) if self.iterparse else ""
+        if not is_list_like(self.iterparse[row_node]):
+            raise TypeError(
+                f"{type(self.iterparse[row_node])} is not a valid type "
+                "for value in iterparse"
+            )
+
+        if (
+            not isinstance(self.path_or_buffer, str)
```

Comment:
tests hit this?

Reply:
Same above. Tests run both etree and lxml parsers with the parser fixture.